lesson14/index.js

```diff
@@ -0,0 +1,92 @@
+const button = document.getElementById("js-button");
+const modalOpenButton = document.getElementById("js-modalOpen");
+const div = document.getElementById("js-contents");
+const modal = document.getElementById("js-modal");
+const ul = document.getElementById("js-lists");
+
+function addLoading() {
+    const img = document.createElement("img");
+    img.src = "loading-circle.gif";
+    img.id = "js-loading";
+    div.appendChild(img);
+}
+
+function removeLoading() {
+    const img = document.getElementById("js-loading");
+    div.removeChild(img);
+}
+
+async function fetchData() {
+    try {
+        const json = await (await fetch("https://myjson.dit.upm.es/api/bins/7ctn")).json();
+        return json.data;
+    } catch (e) {
+        throw new Error(e);
+    }
+}
+
+async function fetchListData() {
+    addLoading();
+    try {
+        const data = await fetchData();
+        if (data.length === 0) {
+            throw new Error("データが空です");
+        }
+        return data;
+    } catch (e) {
+        createErrorMessage(e);
+    } finally {
+        removeLoading();
+    }
+}
+
+function createErrorMessage(e) {
+    const li = document.createElement("li");
+    li.textContent = `エラー内容:${e.message}`;
+    ul.appendChild(li);
+    console.error(e.message);
+}
+
+async function addList() {
+    const fragment = document.createDocumentFragment();
+    const values = await fetchListData();
+
+    if (values) {
+        values.forEach((value) => {
+            const li = document.createElement("li");
+            const anchor = document.createElement("a");
+            const img = document.createElement("img");
+
+            anchor.textContent = value.text;
+            anchor.href = `/${value.to}`;
+            img.src = value.img;
+            img.alt = value.alt;
+
+            li.appendChild(anchor).insertAdjacentElement("afterbegin", img);
+            fragment.appendChild(li);
+        });
+        ul.appendChild(fragment);
+    }
+}
+
+function fetchNumber() {
+    const numberField = document.getElementById("number");
+    const number = numberField.value;
+    
+    if(number) {
+        console.log(number);
+    } else {
+        console.log('null');
+    }
+}
```

Comment:
IMO
I thought it would be better to give them an alert or error message if they did not enter the information.
```suggestion
function fetchNumber() {
  const numberField = document.getElementById("number");
  const number = numberField.value;
 //入力が空だったらアラートを出して処理終了
  if (number === "") {
    alert("入力してください。");
    return;
  }
  console.log(number);
  //ここでaddlistを実行させる
  addList();
}
```
but...
With this process, the modal disappears after the alert is issued and you cannot retype it. So we will need to fix that as well.😂


It's not in the specs of the assignment, so I'll leave it up to you to fix it or not.

Reply:
Thank you for your review!
Thanks for the advice/implementation tips on the alerts!
I fixed below two point.
- Alert me when a number is not entered.
- Make sure that the modal does not disappear when the alert is OK.

4613e4f
[CodeSandbox](https://codesandbox.io/s/javascript-lesson14-c1nvw?file=/index.js)

Reply:
How to write the conditions...

If it is ture in this function
  ```js
closeModal();
addList();
```
If this function is a ture, I guess I could just run

```js
function fetchNumber() {
  const numberField = document.getElementById("number");
  const number = numberField.value;
  if (number === "") {
    alert("入力してください");
    return;
  }
  console.log(number);
  closeModal();
  addList();
}
```
[CodeSandBox](https://codesandbox.io/s/javascript-lesson14-forked-ojdst?file=/index.js)

I made this one a function so that you can see what the purpose of the process is.
```js
function closeModal() {
  const modal = document.getElementById("js-modal");
  modal.classList.remove("is-active");
}

function openModal() {
  const modal = document.getElementById("js-modal");
  modal.classList.add("is-active");
}
```

Reply:
Thank you for your very thorough explanation!
Adding the function made the code very easy to understand just by reading it from above.
I could not have written it this way myself.
Thank you very much!

I fix it.
19f2e31
[CodeSandbox](https://codesandbox.io/s/c1nvw?file=/index.js)